File: proj/CoreLibrary/src/com/jasonzqshen/familyaccounting/core/exception/format/MasterDataFileFormatException.java
package com.jasonzqshen.familyaccounting.core.exception.format;

import com.jasonzqshen.familyaccounting.core.masterdata.MasterDataType;

public class MasterDataFileFormatException extends Exception {
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	public final MasterDataType TYPE;

	public MasterDataFileFormatException(MasterDataType type) {
		super(String.format(
				"Master data file for master data %s contains format error",
				type));
		TYPE = type;
	}
}

File: trunk/proj/WinStore_FamilyAccounting/WinStore_FamilyAccountingCore/Exceptions/FormatExceptions/MasterDataFileFormatException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WinStore_FamilyAccountingCore.MasterData;

namespace WinStore_FamilyAccountingCore.Exceptions.FormatExceptions
{
    public class MasterDataFileFormatException : FormatException
    {
        public readonly MasterDataType TYPE;

        public MasterDataFileFormatException(MasterDataType type)
            : base(String.Format(
                    "Master data file for master data %s contains format error",
                    type))
        {
            TYPE = type;
        }
    }
}
